new/client/src/App.js

```diff
@@ -42,14 +44,20 @@ class App extends Component {
                   this.state.isLoggedin===true ? <Redirect to="/" /> : ( this.state.isLoggedin===false ? <AuthPage {...props} changeState={ this.changeState } /> : null )
                 )} /> */}
                 <Route path="/new" render={props=>(
-                  this.state.isLoggedin===true ? <CreateBlogForm {...props} /> : (this.state.isLoggedin===false ? <Redirect to="/login" /> : null)
+                  this.state.isLoggedin===true ? <CreateBlogForm {...props} /> : (this.state.isLoggedin===false ? <Redirect to="/login" /> : null )
                 )} />
                 <Route path="/login" render={props=>(
                   this.state.isLoggedin===true ? <Redirect to='/' /> : ( this.state.isLoggedin===false ? <Login {...props} changeState={ this.changeState } /> : null )
                 )} />
                 <Route path="/signup" render={props=>(
                   this.state.isLoggedin===true ? <Redirect to='/' /> : ( this.state.isLoggedin===false ? <Register { ...props } /> : null )
                 )} />
+                <Route path="/edit/:username" render={props=>(
+                  this.state.isLoggedin===true ? <Profile {...props} /> : (this.state.isLoggedin===false ? <Redirect to="/login" /> : null )
+                )} />
```

Comment:
How is it making sure that a user cannot edit another user's profile if he is logged in?

Reply:
How will he edit if he does not get another user's information from server, this page will show up if user is loggedin and only for loggedin user. Edit button will be in profile page for only that user whoose profile it is.

Reply:
If anyone loggedin user want to access this page, then it will only show his information not of others. And edit option is placed in profile page and it will be visible only to that use whose profile it is actually

Reply:
I mean, suppose a user `xyz` goes to another user `abc`'s  edity page from directly the browser url section `/edit/abc`, can he edit it?

Reply:
You can check it in your system, it will work properly

Reply:
> I mean, suppose a user `xyz` goes to another user `abc`'s edity page from directly the browser url section `/edit/abc`, can he edit it?

No he can't edit instead it will show up his own information as parsed token gives the id loggedin user not of url params

Reply:
I have taken user id from token not from url params